bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -386,129 +375,73 @@ def __call__(self, inputs):
-        self._predict_fn_name = predict_fn_name
-        self._partial_kwargs: t.Dict[str, t.Any] = (
-            partial_kwargs if partial_kwargs is not None else dict()
+        tf.saved_model.save(
+            model,
+            _model.path,
+            signatures=tf_signatures,
+            options=tf_save_options,
         )
 
-    def _configure(self, device_id: str) -> None:
-        if "GPU" in device_id:
-            tf.config.set_visible_devices(device_id, "GPU")
-        self._config_proto = dict(
-            allow_soft_placement=True,
-            log_device_placement=False,
-            intra_op_parallelism_threads=self._num_threads,
-            inter_op_parallelism_threads=self._num_threads,
-        )
-
-    @property
-    def _num_threads(self) -> int:
-        if is_gpu_available() and self.resource_quota.nvidia_gpu:
-            return 1
-        return max(round(self.resource_quota.cpu), 1)
-
-    @property
-    def num_replica(self) -> int:
-        if is_gpu_available() and self.resource_quota.nvidia_gpu:
-            return self.resource_quota.nvidia_gpu
-        return 1
-
-    def _setup(self) -> None:
-        self._model = load(self._tag, model_store=self.model_store)
-        raw_predict_fn = getattr(self._model, self._predict_fn_name)  # type: ignore
-        self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
-
-    def _run_batch(self, *args: "TFArgType", **kwargs: "TFArgType") -> "ext.NpNDArray":
-        params = Params["TFArgType"](*args, **kwargs)
-
-        with tf.device(self._device_id):  # type: ignore
-
-            def _mapping(item: "TFArgType") -> "tf_ext.TensorLike":
-                if not LazyType["tf_ext.TensorLike"]("tf.Tensor").isinstance(item):
-                    return t.cast("tf_ext.TensorLike", tf.convert_to_tensor(item))
-                else:
-                    return item
-
-            params = params.map(_mapping)
+        return _model.tag
 
-            tf.compat.v1.global_variables_initializer()  # type: ignore
-            res = self._predict_fn(*params.args, **params.kwargs)
-            return t.cast("ext.NpNDArray", res.numpy())
 
+def _to_runnable(
+    bentomodel: BentoModel,
+    tags: t.Optional[t.List[str]] = None,
+    load_as_hub_module: t.Optional[bool] = None,
+    partial_kwargs: t.Dict[str, Any] = None,
+    **kwargs,
+):
+    class TensorflowRunnable(Runnable):
+        SUPPORT_GPU: True
+        SUPPORT_MULTI_THREADING: True
+
+        def __init__(self):
+            super().__init__(**kwargs)
+            if tf.list_physical_devices("GPU") > 0:
+                self.device_id = "/device/GPU:0"
+            else:
+                self.device_id = "/device/CPU:0"
+                self.num_threads = int(os.environ["BENTOML_NUM_THREAD"])
+                tf.config.threading.set_inter_op_parallelism_threads(self.num_threads)
+                tf.config.threading.set_intra_op_parallelism_threads(self.num_threads)
```

Comment:
move this to _load_model instead and remove the device_id params from it?

Reply:
`load_model` let user test the save/load process and I think it's convenient to let they choose which device the model should be on